src/main/java/com/jcabi/log/ColorManager.java

```diff
@@ -0,0 +1,97 @@
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.log;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Color manager to store color data.
+ * @author Igor Piddubnyi (igor.piddubnyi@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public class ColorManager {
+
+    /**
+     * Colors with names.
+     */
+    private final transient ConcurrentMap<String, String> colors =
+        ColorManager.colorMap();
+
+    /**
+     * Add color to color map.
+     * @param key Key to add.
+     * @param value Value to add.
+     */
+    public final void addColor(final String key, final String value) {
+        this.colors.put(key, value);
+    }
+
+    /**
+     * Convert our text to ANSI color.
+     * @param meta Meta text
+     * @return ANSI color
+     */
+    public final String ansi(final String meta) {
+        final String ansi;
+        if (meta == null) {
+            ansi = "?";
+        } else if (meta.matches("[a-z]+")) {
+            ansi = this.colors.get(meta);
+            if (ansi == null) {
+                throw new IllegalArgumentException(
+                    String.format("unknown color '%s'", meta)
+                );
+            }
+        } else {
+            ansi = meta;
+        }
+        return ansi;
+    }
+
+    /**
+     * Color map.
+     * @return Map of colors
+     */
+    private static ConcurrentMap<String, String> colorMap() {
+        final ConcurrentMap<String, String> map =
+            new ConcurrentHashMap<String, String>();
```

Comment:
@piddubnyi We're not modifying that map after creation, I think we should use `ImmutableMap` instead